lib/__tests__/fixtures/plugin-meta-fixer.mjs

```diff
@@ -0,0 +1,48 @@
+import stylelint from '../../index.mjs';
+
+const {
+	createPlugin,
+	utils: { report, ruleMessages, validateOptions },
+} = stylelint;
+
+const ruleName = 'plugin/selector-no-foo';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (selector) => `Unexpected "foo" in selector "${selector}"`,
+});
+
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [true],
+		});
+
+		if (!validOptions) return;
+
+		root.walkRules((ruleNode) => {
+			const { selector } = ruleNode;
+			const i = selector.indexOf('foo');
+
+			if (i === -1) return;
+
+			report({
+				result,
+				ruleName,
+				message: messages.rejected(selector),
+				node: ruleNode,
+				index: i,
+				endIndex: i + 'foo'.length,
+				fix: () => {
+					ruleNode.selector = selector.replace('foo', 'bar');
+				},
+			});
+		});
+	};
+};
+
+rule.ruleName = ruleName;
+rule.messages = messages;
+rule.meta = { fixer: true };
```

Comment:
`fixer` is not needed right at a functional level, right?

Should we omit it altogether?

Reply:
I thought about it. It depends on how you want the granularity of the warning to be for unmigrated plugins.
i.e. either we trust the `fixer` property or we assume trickery from plugin authors and add a getter to `context.fix`

What do you reckon?

Reply:
@ybiquitous what's your take on this?
i.e. what should trigger a warning/output for plugins that haven't been migrated yet.
e.g. when the formatter is verbose we could have a count of the outdated plugins and a link to the documentation

Reply:
This new `fixer` property would be temporary, right? If all plugins migrated to the new API, would this property become unnecessary?

Also, I wonder if plugin authors would feel annoyed to have to add `fixer`.

If the suggested `context.fix` change works, I am in favor of `context.fix`.

Reply:
> This new fixer property would be temporary, right? If all plugins migrated to the new API, would this property become unnecessary?

The real benefit would be for us to know **without** linting if some outdated rules are present in the resolved config.
i.e. without `fixer` we wouldn't be able to implement a `--outdated` flag to list those rules

> I am in favor of `context.fix`

If @romainmenke agrees too Ill remove it.

Reply:
But, having to specify both `fix` and `fixer` doesn't make sense, right?

Reply:
If the name the problem, it can be changed. Anyway I thought having that information in `meta` would be more versatile.
Ill just remove it for now; it's not needed for that particular PR anyway.